docs/source/ts-ssdk/error-handling.rst

```diff
@@ -0,0 +1,75 @@
+#####################################################
+Smithy Server Generator for TypeScript error handling
+#####################################################
+
+Errors are structures associated with services and operations in the Smithy model and are used to indicate that the
+server encountered a problem during request processing. Structures with the error trait are code generated as subclasses
+of ``Error``.
+
+If a code-generated ``Error`` is thrown from a handler implementation, and the ``Error`` is associated with the
+operation or service in the model, then the server SDK will serialize the Error according to the rules of the protocol.
+For instance, HTTP binding protocols set the status code of the response based on the ``@httpError`` trait.
+
+If a non-code-generated Error, or a code-generated Error that is not associated with the operation or service in the
+model, is thrown from a handler, then a *synthetic* ``InternalFailure`` will be rendered as the result instead.
+
+Synthetic errors
+================
+
+*Synthetic errors* are errors that are not included in the Smithy model, but can still be thrown by the server SDK. In
+general, these errors are not expected to have corresponding code generated types on the client side. These errors fall
+into two categories: framework-level errors that are unavoidable, and errors that are associated with the low-level
```

Comment:
The default errors from API Gateway should also have its own category as part of the Synthetic errors. With the difference that the service team will have no control over it (as explained briefly in [this issue](https://github.com/awslabs/smithy/issues/1170)).

Reply:
I think those would need their own section once we come up with a more specific API Gateway solution.

There are always going to be intermediaries (CDNs, caches, reverse proxies, what-have-you) that can return error responses that the server has no control over. While we have some customization options for API Gateway, generated clients should be durable to these failures, and I don't think the proper experience is customers having a modeled type for things like `502 Bad Gateway`.